### ISSUE TITLE
Adding new/additional categories and attributes for SCADA/ICS issues/events/incidents

### ISSUE THREAD

**Issue by infracritical** [MISP/MISP#965]
Our organization does research pertaining to critical infrastructure cybsecurity, specifically with SCADA and control systems.  The requirements of reporting and identifying events and/or incidents is much different from traditional IT-based issues.

I would like to customize the MISP database to allow for these modifications.  If necessary, I can provide a schema of what are the requirements.  Outside of the "addAttribute.sh" script, are there any features available that would allow such a function to exist?  If not, how difficult would it be to create one?

**Comment by iglocska** [comment]
Add attribute is something different, it's a sample script on how to use
curl to add an attribute of one of the existing types.

However, we are always interested in use cases that out current data model
doesn't cover. There is an issue where we track requests for new attribute
types, so i think the best would be if you could describe the attributes
that you'd need over there. Sadly, i am currently on my phone and cannot
search for it, but i will link it in this issue asap.

**Comment by adulau** [comment]
As @iglocska mentioned, we are very interested in existing ICS use-cases (and your schema) that would need new attributes or new combined objects. 

Regarding the combined objects, the work-in-progress is located here:

https://github.com/MISP/misp-objects

Cheers

**Comment by infracritical** [comment]
But...wouldn't this make better sense to have this feature to add/edit categories and attributes?

I have no issue sharing the data schema; however, this is _all_ uncharted waters, and I still have the issue with dealing with the SCADA community (henceforth, why I asked about being capable of adding or editing the categories and attributes).

In the meantime, I can see about providing an initial data schema for starters.

**Comment by adulau** [comment]
It's a very good question but we usually try to have common attributes that people/organisations actually use.

As we would like to have attributes and categories that people can share easily (especially having common meaning and interpretation). The whole model of MISP is built around the synchronisation, sharing and contributing around a network of MISP instances. Having a whole set of common attributes ICS/SCADA would make sense as this is an interesting use case. If you have an initial data schema, we could have a look at it.

Thank you for your feedback.

**Comment by infracritical** [comment]
Absolutely.  I can have something put together within the next week (or so) - time permitting.  ;)

**Comment by Rafiot** [comment]
@infracritical any news?

**Comment by infracritical** [comment]
I apologize that I have not had any time.  NERC CIP v5 goes 'live' on 1-Apr-2016, so things are kind of tense.  Once the regulatory deadline has passed, I should have some more time to address this issue.

**Comment by adulau** [comment]
Let us know if you have more info. I'll close the issue but feel free to reopen it when you get more information.

**Comment by infracritical** [comment]
I apologize for never responding from earlier.  Between work and major health issues, been a tad bit tied up.  This effort will begin resuming again with building up taxonomies for SCADA and control systems.

**Comment by adulau** [comment]
@infracritical Great news! If you have already some pointers or reference, we could open an issue on the taxonomy to focus on the ICS/SCADA aspects. https://github.com/MISP/misp-taxonomies - https://www.misp-project.org/taxonomies.html

**Comment by iglocska** [comment]
Health comes first, glad to have you back ;)
